### ISSUE TITLE
Boosts only visible to my followers still announced to original poster

### ISSUE THREAD

**Issue by FremyCompany** [mastodon/mastodon#21295]
### Steps to reproduce the problem

0. Create or connect to a profile you control (Account0)

1. Create an account with protected follows (Account1)
2. Change its post privacy to "followers only"
3. Boost a tweet from Account0, which does not follow you
4. Account1 receives a notification that you boosted its message, but cannot access that reblog normally

I think this is not the expected behavior. 

### Expected behaviour

The reboost notification should **not** be sent in this case; it should only be sent if Account0 were to follow Account1 and be able to see the activity.

### Actual behaviour

Account0 receives a notification of Account1's reblog, depite not being able to see it.

### Detailed description

For my use case, this is detrimental because this notification is an unwanted noise for the original poster.
But, more generally, I can imagine this is a leak of privacy for Account1. They do not expect the original poster to be notified of the messages they post, even reboosts.

### Specifications

Mastodon v4.0.2

**Comment by straithe** [comment]
I feel that account0 being able to see that their content is being boosted by private accounts is a safety feature to them. If a LOT of private accounts are boosting their content it may help them be aware of incoming targeted attacks if account1 were to be malicious towards account0 or a community/group they belong to.

**Comment by FremyCompany** [comment]
This also means that the reboosting activity of a private user is potentially being sent to another server they don't control, giving the admins of that server precise information about what messages that private user saw and when they were active on mastodon (and other metadata part of the ANNOUNCE activity). The main intent of a private user is to not leak information to non-followers, which can be hand-picked to ensure they are safe to interact with.

I feel like "what if there are a lot of private accounts boosting you" is not a very strong use case. Bad actors can anyway just take a screenshot and share that if they don't want to create noise.

The end result of this is that reboosting is less safe for private accounts, which I think is worse than the alternative.

**Comment by ineffyble** [comment]
Account0 is already able to see that they are being followed by Account1 in this case, right?

**Comment by FremyCompany** [comment]
Yes, although you can boost tweets from people you do not follow, if they arrived on your timeline via a boost from someone you follow instead.
